source/index.html.haml

```diff
@@ -75,7 +75,9 @@ hide_metadata: true
           = data.site.name
           News
 
-        = partial :blog_posts, locals: locals
+        -### Load last 3 blog entry summaries ###
+        - page_articles[0..2].reject {|a| !a.published? }.each do |article|
+          = partial :blog_post, locals: {article: article, summarize: 1}
```

Comment:
...Or, of course, the `source/layouts/blog_posts.haml` could be modified to support a # of articles for the front page too, so you don't have to do any of this here.

(This would be the better approach.)